### PR TITLE
feat: global state root computation

### PR DESCRIPTION
**Pull request by ben2077**
…e root, and global state root.

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing
- [ ] Other (please describe):

# What is the current behavior?

Issue Number: N/A

# What is the new behavior?

-
-
-

# Does this introduce a breaking change?

- [ ] Yes
- [ ] No

# Other information

### CODE REVIEW

File: crates/primitives/starknet/src/crypto/commitment/mod.rs [hunk 1]
```diff
@@ -153,3 +160,81 @@ pub fn calculate_event_hash<T: CryptoHasher>(event: &EventWrapper) -> FieldEleme
...
+}
+
+/// Calculate the hash of a contract class.
+fn calculate_contract_class_hash<T: CryptoHasher>(class: &ClassHash) -> FieldElement {
+    let class_hash = FieldElement::from_byte_slice_be(class.0.bytes()).unwrap();
+    T::compute_hash_on_elements(&[FieldElement::from_byte_slice_be(CONTRACT_CLASS_LEAF_V0).unwrap(), class_hash])
+}
+
+/// Calculate the global state root.
+pub fn calculate_global_state_root<T: CryptoHasher>(contract_state_root: H256, classes_root: H256) -> H256 {
```
Comment (abdelhamidbakhta) [inline crates/primitives/starknet/src/crypto/commitment/mod.rs:231]: please add rust doc comments (arguments, returns)

**Review by abdelhamidbakhta — CHANGES_REQUESTED** [review]
left comments

**Comment by abdelhamidbakhta** [comment]
hey @ben2077 what's the status of this PR ?

**Comment by LucasLvy** [comment]
Do you need help for this ?

**Pull request closed without merging**